src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -462,6 +505,54 @@ int encrypt(byte[] in, int inOfs, int len, byte[] out, int outOfs) {
         return len;
     }
 
+    int decrypt(ByteBuffer src, ByteBuffer dst) {
+        if (src.remaining() > 0) {
+            byte[] b = new byte[src.remaining()];
+            src.get(b);
+            try {
+                ibuffer.write(b);
+            } catch (IOException e) {
+                throw new ProviderException("Unable to add remaining input to the buffer", e);
+            }
+        }
+        return 0;
+    }
+
+
+    int encrypt(ByteBuffer src, ByteBuffer dst) {
+        int len = src.remaining();
+        if (len == 0) {
+            return 0;
+        }
+
+        ArrayUtil.blockSizeCheck(src.remaining(), blockSize);
```

Comment:
Hmm, I am not sure if this check still applies in ByteBuffer case. You are passing the ByteBuffer objs directly from AESCipher->CipherCore->GaloisCounterMode. This is different from the byte[] case where CipherCore would chop up the data into blocks and pass the blocks to the underlying FeedbackCipher impl. Perhaps no existing regression tests covers ByteBuffer inputs w/ non-blocksize data? Otherwise, this should be caught?
BTW, why not just use 'len' again? Seems unnecessary to keep calling src.remaining() in various places in this method.

Reply:
Yes the check is unnecessary

I suspect not using len was simply a mistake